### PR TITLE
hotfix/set-root-weights

### PR DESCRIPTION
**Pull request by JohnReedV**
## Description
This PR hotfixes a bug that occurred when setting root weights. When checking if the caller had the minimum stake the coldkey was used rather than the hotkey. 

```{'code': 1010, 'message': 'Invalid Transaction', 'data': 'Transaction call is not expected'}```

## TODO
BUMP SPEC VERSION

### CODE REVIEW

**Review by sam0x17 — DISMISSED** [review]

**Review by open-junius — DISMISSED** [review]

**Review by sam0x17 — DISMISSED** [review]

**Comment by open-junius** [comment]
we can merge it, PR in bittensor https://github.com/opentensor/bittensor/pull/2021 will fix the failed test case.

**Review by distributedstatemachine — DISMISSED** [review]

**Review by camfairchild — APPROVED** [review]

File: pallets/subtensor/tests/block_step.rs [hunk 1]
```diff
@@ -869,7 +869,9 @@ fn test_emission_based_on_registration_status() {
             n as usize
         );
 
+        let block: u64 = 0;
         // drain the emission tuples for the subnet with registration on
+        #[allow(clippy::fn_to_numeric_cast)]
         SubtensorModule::drain_emission(next_block as u64);
```
Comment (camfairchild) [inline pallets/subtensor/tests/block_step.rs]:
```suggestion
        SubtensorModule::drain_emission(block as u64);
```

File: pallets/subtensor/tests/block_step.rs [hunk 2]
```diff
@@ -869,7 +869,9 @@ fn test_emission_based_on_registration_status() {
             n as usize
         );
 
+        let block: u64 = 0;
         // drain the emission tuples for the subnet with registration on
+        #[allow(clippy::fn_to_numeric_cast)]
```
Comment (camfairchild) [inline pallets/subtensor/tests/block_step.rs]:
```suggestion
```

Comment (camfairchild) [reply pallets/subtensor/tests/block_step.rs]: we shouldn't be casting any functions

File: pallets/subtensor/tests/block_step.rs (hunk 1 above)
Comment (camfairchild) [reply pallets/subtensor/tests/block_step.rs]: perhaps this was meant to be block

**Review by camfairchild — DISMISSED** [review]

**Review by camfairchild — APPROVED** [review]

**Review by distributedstatemachine — APPROVED** [review]

**Review by sam0x17 — APPROVED** [review]

**Pull request merged**